frontend/workflows/dynamodb/src/update-capacity.tsx

```diff
@@ -0,0 +1,254 @@
+          <TableRow key={table.name}>
+            {table.name}
+            Table
+            {table.status}
+            <MetadataTable
+              onUpdate={handleTableCapacityChange}
+              data={[
+                {
+                  name: "read",
+                  value: table.provisionedThroughput.readCapacityUnits,
+                  input: {
+                    type: "number",
+                    key: "read",
+                    validation: number()
+                      .integer()
+                      .transform(value => (Number.isNaN(value) ? 0 : value))
+                      .min(Number(table.provisionedThroughput.readCapacityUnits)),
+                  },
+                },
+                {
+                  name: "write",
+                  value: table.provisionedThroughput.writeCapacityUnits,
+                  input: {
+                    type: "number",
+                    key: "write",
+                    validation: number()
+                      .integer()
+                      .transform(value => (Number.isNaN(value) ? 0 : value))
+                      .min(Number(table.provisionedThroughput.writeCapacityUnits)),
+                  },
+                },
+              ]}
+            />
+          </TableRow>
+          {table.globalSecondaryIndexes.map(gsi => (
+            <TableRow key={gsi.name}>
+              {gsi.name}
+              Index
+              {gsi.status}
+              <MetadataTable
+                onUpdate={handleGsiCapacityChange}
+                data={[
+                  {
+                    name: "read",
+                    value: gsi.provisionedThroughput.readCapacityUnits,
+                    input: {
+                      type: "number",
+                      key: `read_capacity_units,${gsi.name}`,
+                      validation: number()
+                        .integer()
+                        .transform(value => (Number.isNaN(value) ? 0 : value))
+                        .min(Number(gsi.provisionedThroughput.readCapacityUnits)),
+                    },
+                  },
+                  {
+                    name: "write",
+                    value: gsi.provisionedThroughput.writeCapacityUnits,
+                    input: {
+                      type: "number",
+                      key: `write_capacity_units,${gsi.name}`,
+                      validation: number()
+                        .integer()
+                        .transform(value => (Number.isNaN(value) ? 0 : value))
+                        .min(Number(gsi.provisionedThroughput.writeCapacityUnits)),
+                    },
+                  },
+                ]}
+              />
+            </TableRow>
+          ))}
+        </Table>
+      </Container>
+
+      {/* TO DO: conditionally render the override checkbox depending on workflow config prop */}
+      <CheckboxPanel
+        header="To override the safety limits for scaling, check the box below."
```

Comment:
This could be confusing since the input validation is checking that the min is gte the current value. I understand this is overriding the total capacity limit but only because I read the code here.

Reply:
Revised and reformatted to clarify the warning message 👍